metabase/metabase
# Replace redux-form in reference forms

Epic https://github.com/metabase/metabase/issues/22372

Replaces `redux-form` in the following forms:
- `DatabaseDetail`
- `FieldDetail`
- `TableDetail`
- `SegmentDetail`
- `SegmentFieldDetail`
- `FieldList`
- `SegmentFieldList`

Please note that I needed to convert these components to function components. [Ignore whitespace to see the actual changes](https://github.com/metabase/metabase/pull/24722/files?diff=split&w=1).

How to test:
- Go to `/reference`
- Change entities of specified types and make sure it works

<img width="1728" alt="Screenshot 2022-08-11 at 15 26 33" src="https://user-images.githubusercontent.com/8542534/184133024-0c6780e4-7cb3-4c45-a397-6e0218f30a14.png">
